sources/Outputs/src/Par.cpp

```diff
@@ -0,0 +1,174 @@
+//
+// Copyright (c) 2020, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+
+/**
+ * @file  Par.cpp
+ *
+ * @brief Dynaflow launcher PAR file writer implementation file
+ *
+ */
+
+#include "Par.h"
+
+#include "Constants.h"
+
+#include <PARParameter.h>
+#include <PARParameterFactory.h>
+#include <PARParametersSetCollection.h>
+#include <PARParametersSetCollectionFactory.h>
+#include <PARParametersSetFactory.h>
+#include <PARReference.h>
+#include <PARReferenceFactory.h>
+#include <PARXmlExporter.h>
+#include <boost/filesystem.hpp>
+
+namespace dfl {
+namespace outputs {
+
+namespace helper {
+
+template<class T>
+static boost::shared_ptr<parameters::Parameter>
+buildParameter(const std::string& name, const T& value) {
+  return parameters::ParameterFactory::newParameter(name, value);
+}
+
+static boost::shared_ptr<parameters::Reference>
+buildReference(const std::string& name, const std::string& origName, const std::string& type) {
+  auto ref = parameters::ReferenceFactory::newReference(name);
+  ref->setOrigData("IIDM");
+  ref->setOrigName(origName);
+  ref->setType(type);
+
+  return ref;
+}
+
+}  // namespace helper
+
+Par::Par(ParDefinition&& def) : def_{std::forward<ParDefinition>(def)} {}
+
+void
+Par::write() {
+  parameters::XmlExporter exporter;
+
+  auto collection = parameters::ParametersSetCollectionFactory::newCollection();
+
+  auto constants = writeConstantSets(def_.generators.size());
+  for (auto it = constants.begin(); it != constants.end(); ++it) {
+    collection->addParametersSet(*it);
+  }
+
+  for (auto it = def_.generators.begin(); it != def_.generators.end(); ++it) {
+    auto set = writeGenerator(*it, def_.basename);
+    if (set) {
+      collection->addParametersSet(set);
+    }
+  }
+
+  exporter.exportToFile(collection, def_.filepath);
+}
+
+void
+Par::updateSignalNGenerator(boost::shared_ptr<parameters::ParametersSet> set) {
+  set->addParameter(helper::buildParameter("generator_KGover", 1.));
+  set->addParameter(helper::buildParameter("generator_QMin", -constants::qValueMax));
+  set->addParameter(helper::buildParameter("generator_QMax", constants::qValueMax));
+  set->addParameter(helper::buildParameter("generator_PMin", -constants::qValueMax));
```

Comment:
Please rename the constant if it is also used for active power

Reply:
OK